File: Backend/src/models/article-model.ts
import { Articles } from "@prisma/client"

export interface CreateArticleRequest {
    title: string
    text: string
    image_url?: string
    game_id: number
}

export interface ArticleResponse {
    article_id: number
    title: string
    text: string
    image_url: string | null
    game_id: number
    createdDate: Date
}

export function toArticleResponse(article: Articles): ArticleResponse {
    return {
        article_id: article.article_id,
        title: article.title,
        text: article.text,
        image_url: article.image_url,
        game_id: article.game_id,
        createdDate: article.createdDate
    }
}

File: Frontend/app/src/main/java/com/visprog/starchive/models/ArticleModel.kt
package com.visprog.starchive.models

data class GetAllArticleResponse(
    val data: List<ArticleModel>
)

data class GetArticleResponse(
    val data: ArticleModel
)

data class ArticleModel(
    val article_id: Int,
    val title: String,
    val text: String,
    val image_url: String?,
    val game_id: Int,
    val createdDate: String
)

data class ArticleRequest(
    val article_id: Int,
    val title: String,
    val text: String,
    val image_url: String?,
    val game_id: Int,
    val createdDate: String
)